Source/Core/Core/PowerPC/GDBStub.cpp

```diff
@@ -31,6 +32,29 @@ typedef SSIZE_T ssize_t;
 #include "Core/PowerPC/PPCCache.h"
 #include "Core/PowerPC/PowerPC.h"
 
+namespace
+{
+#ifdef _WIN32
+class WSA_RAII
+{
+public:
+  WSA_RAII() { (void)WSAStartup(MAKEWORD(2, 2), &m_data); }
+  ~WSA_RAII() { WSACleanup(); }
+
+  WSA_RAII(const WSA_RAII&) = delete;
+  WSA_RAII(WSA_RAII&&) = delete;
+
+  WSA_RAII& operator=(const WSA_RAII&) = delete;
+  WSA_RAII& operator=(WSA_RAII&&) = delete;
+
+private:
+  WSADATA m_data;
+};
+
+std::unique_ptr<WSA_RAII> s_wsa;
```

Comment:
A unique_ptr is kind of wasteful for this -- a std::optional would suffice if all you want is being able to construct/destruct at will.

Reply:
Heh, I almost did that.  I just wasn't sure what would be better